src/main/java/org/snomed/snowstorm/rest/ConceptController.java

```diff
@@ -410,10 +410,11 @@ public ResponseEntity<ConceptView> createConcept(
 			@RequestHeader(value = "Accept-Language", defaultValue = Config.DEFAULT_ACCEPT_LANG_HEADER) String acceptLanguageHeader) throws ServiceException {
 		branch = BranchPathUriUtil.decodePath(branch);
 		Concept concept = (Concept) conceptView;
+		// Remove temporary ids before the underlying create operation.
+		concept = ConceptValidationHelper.stripTemporaryUUIDsIfSet(concept);
```

Comment:
I found I had to shift this logic up (outside the if-block) as my unit test was failing due to some identifier issue.

Reply:
@dmcgihtsdo The unit test failed because you have set the concept id with uuid in the buildConceptJson() method.

Reply:
Thanks @CoderMChu .

I have removed the id from the buildConceptJson method and dropped the commit shifting the logic. The tests still pass.